io-engine/src/bdev/nexus/nexus_bdev_rebuild.rs

```diff
@@ -177,29 +177,27 @@ impl<'n> Nexus<'n> {
 
     /// Translates the job into a new history record and pushes into
     /// the history.
-    fn create_history_record(mut self: Pin<&mut Self>, job: Arc<RebuildJob>) {
+    fn create_history_record(&self, job: Arc<RebuildJob>) {
         let Some(rec) = job.history_record() else {
             error!("{self:?}: try to get history record on unfinished job");
             return;
         };
 
-        unsafe {
-            self.as_mut().get_unchecked_mut().rebuild_history.push(rec);
-        }
+        self.rebuild_history.lock().push(rec);
 
         debug!(
             "{self:?}: new rebuild history record for '{dst}'; \
             total {num} records",
             dst = job.dst_uri,
-            num = self.rebuild_history.len()
+            num = self.rebuild_history.lock().len()
```

Comment:
nit: should we get the len before the debug! to avoid taking the lock while writing to stdout? probably doesn't matter too much though

Reply:
isn't the arg expr calculated before calling a func?